perf-tool/src/server.cpp

```diff
@@ -109,133 +143,178 @@ void Server::handleServer()
         pollFds[0].revents = 0;
     }
     
-    if (verbose >= WARN)
+    if (verbose >= Verbose::WARN)
         printf("Server started.\n");
 
-    /* Use polling to keep track of clients and keyboard input */
-    while (keepPolling)
+    /* Use polling to keep track of network clients */
+    if (serverSocketFd > 0)
     {
-        if (serverSocketFd >= 0)
+        while (keepPolling)
         {
-            if (poll(pollFds, activeNetworkClients + ServerConstants::BASE_POLLS, ServerConstants::POLL_INTERVALS) == -1)
+            int pollTimeMillis = ServerConstants::POLL_INTERVALS;
+            if (!delayedCommands.empty())
+            {
+                int nextOperationDelay = executeAllDueCommands();
+                if (nextOperationDelay > 0)
+                    pollTimeMillis = std::min(nextOperationDelay*1000, pollTimeMillis);
+            }
+            int pollResult = poll(pollFds, activeNetworkClients + ServerConstants::BASE_POLLS, pollTimeMillis);
+            if (pollResult == -1)
             {
                 perror("ERROR on polling, server will now shut down");
                 break;
             }
 
-            if (activeNetworkClients < ServerConstants::NUM_CLIENTS && (pollFds[0].revents & POLLIN))
+            if (pollResult > 0) /* data is ready on at least one socket */
             {
-                /* The accept() call actually accepts an incoming connection */
-                clilen = sizeof(cli_addr);
-
-                /* This accept() function will write the connecting client's address info 
-                 *into the the address structure and the size of that structure is clilen.
-                 */
-                newsocketFd = accept(serverSocketFd, (struct sockaddr *)&cli_addr, &clilen);
-                if (newsocketFd < 0)
-                    perror("ERROR on accept");
-                else
-                    networkClients[activeNetworkClients] = new NetworkClient(newsocketFd);
-                
-                if (verbose == INFO)
-                    printf("server: got connection from %s port %d\n",
-                       inet_ntoa(cli_addr.sin_addr),
-                       ntohs(cli_addr.sin_port));
-
-                /* Send a welcome message */
-                sendMessage(newsocketFd, "Connection to server succeeded", "ServerStartEvent", "Server");
-
-                /* Update number of active clients */
-                activeNetworkClients++;
+                /* First check for incoming commands from network clients */
+                for (int i = 0; i < activeNetworkClients; i++)
+                {
+                    int socketIndex = ServerConstants::BASE_POLLS + activeNetworkClients;
+                    if (pollFds[socketIndex].revents & (POLLERR | POLLHUP | POLLNVAL))
```

Comment:
Is it possible for this to change after the fd has been set to -1? May be worth guarding this if/elseif with 
```
if (pollFds[socketIndex].fd > 0)
```

Reply:
I added a check to skip over sockets that are closed.